src/udx.c

```diff
@@ -2195,26 +2262,50 @@ udx_stream_write_resume (udx_stream_t *stream, udx_stream_drain_cb drain_cb) {
   return 0;
 }
 
+static void
+_udx_stream_write (udx_stream_write_t *write, udx_stream_t *stream, const uv_buf_t bufs[], unsigned int bufs_len, udx_stream_ack_cb ack_cb, bool is_write_end) {
+  assert(bufs_len > 0);
+
+  // initialize write object
+
+  write->size = 0;
+  write->bytes_acked = 0;
+  write->is_write_end = is_write_end;
+  write->stream = stream;
+  write->on_ack = ack_cb;
+
+  // consider:
+  // make `udx_stream_write` re-entrant, allowing write to be called again to add more buffers to the same request object
+
+  // can't create the buffers in a block because it is hard to determine where to free them:
+  // the write request object is owned by the user, we can't hook into it's destruction
+  // the stream_t object could hold writes but they may grow indefinitely
+
+  for (unsigned int i = 0; i < bufs_len; i++) {
+    udx_stream_write_buf_t *wbuf = calloc(1, sizeof(*wbuf)); // calloc to zero wbuf
```

Comment:
could we move this allow to the udx_stream_write_t's creator? ie when the user allocs a write req the allow the array also. then upstream we can reuse them for all the simple single writes? wdyt?